File: tests/bench/bench.c
#include <stdio.h>
#include <math.h>
#include <stdbool.h>

bool eh_primo(int n)
{
    if (n < 2)
        return false;
    if (n == 2)
        return true;
    if (n % 2 == 0)
        return false;

    int limite = (int)sqrt(n);
    for (int i = 3; i <= limite; i += 2)
    {
        if (n % i == 0)
            return false;
    }
    return true;
}

int contar_primos(int limite)
{
    int contador = 0;
    for (int numero = 2; numero <= limite; numero++)
    {
        if (eh_primo(numero))
            contador++;
    }
    return contador;
}

int main()
{
    int limite = 1000000;

    int resultado = contar_primos(limite);
    printf("RESULT:%d\n", resultado);

    return 0;
}

File: tests/bench/bench.go
package main

import (
    "fmt"
    "math"
)

func ehPrimo(n int) bool {
    if n < 2 { return false }
    if n == 2 { return true }
    if n % 2 == 0 { return false }
    
    limite := int(math.Sqrt(float64(n)))
    for i := 3; i <= limite; i += 2 {
        if n % i == 0 { return false }
    }
    return true
}

func contarPrimos(limite int) int {
    contador := 0
    for numero := 2; numero <= limite; numero++ {
        if ehPrimo(numero) {
            contador++
        }
    }
    return contador
}

func main() {
    limite := 1000000
    
    resultado := contarPrimos(limite)
    fmt.Printf("RESULT:%d\n", resultado)
}
